### PR TITLE
Minor things

### PR DESCRIPTION
**Pull request by basz**
bout done with cleanup

### CODE REVIEW

**Comment by basz** [comment]
and the last minor thing php71 messes up my evening... :) bedtime

File: src/Grant/AuthorizationServerAwareTrait.php [hunk 1]
```diff
@@ -32,7 +35,7 @@
     private $authorizationServer;
 
     /**
-     * {@inheritDoc}
```
Comment (prolic) [inline src/Grant/AuthorizationServerAwareTrait.php]: Just remove PHP Doc here

File: src/Model/AbstractToken.php [hunk 2]
```diff
@@ -106,7 +109,7 @@ protected static function createNew(
 
     /**
      * @param array $data
-     * @return AbstractToken
+     * @return static
```
Comment (prolic) [inline src/Model/AbstractToken.php]: Remove PHP doc and use return type

File: src/Model/AuthorizationCode.php [hunk 3]
```diff
@@ -60,9 +63,9 @@ public static function createNewAuthorizationCode(
 
     /**
      * @param array $data
-     * @return AbstractToken
+     * @return AuthorizationCode
```
Comment (prolic) [inline src/Model/AuthorizationCode.php]: Again PHP doc not required

**Comment by prolic** [comment]
Ok seems like conflict is already fixed. Still failing build.

**Comment by prolic** [comment]
Can you find out why PHP 7.1 is failing?

**Comment by basz** [comment]
PHP71RC2 is problematic it seems, wait a bit

File: README.md [hunk 4]
```diff
@@ -3,8 +3,8 @@
 [![Build Status](https://travis-ci.org/zf-fr/zfr-oauth2-server.png)](https://travis-ci.org/zf-fr/zfr-oauth2-server)
 [![Latest Stable Version](https://poser.pugx.org/zfr/zfr-oauth2-server/v/stable.png)](https://packagist.org/packages/zfr/zfr-oauth2-server)
 [![Coverage Status](https://coveralls.io/repos/zf-fr/zfr-oauth2-server/badge.png)](https://coveralls.io/r/zf-fr/zfr-oauth2-server)
-[![Scrutinizer Quality Score](https://scrutinizer-ci.com/g/zf-fr/zfr-oauth2-server/badges/quality-score.png?s=be36235c9898cfc55044f58d9bba789d2d4d102e)](https://scrutinizer-ci.com/g/zf-fr/zfr-oauth2-server/)
 [![Total Downloads](https://poser.pugx.org/zfr/zfr-oauth2-server/downloads.png)](https://packagist.org/packages/zfr/zfr-oauth2-server)
+[![Gitter](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/prolic/zfc-rbac)
```
Comment (prolic) [inline README.md]: wrong url!

File: build/AuthorizationServerInterface.php.html [hunk 5]
```diff
@@ -0,0 +1,199 @@
+<!DOCTYPE html>
```
Comment (prolic) [inline build/AuthorizationServerInterface.php.html]: add to .ignore and remove from PR, also better rebase the branch, so it's not part of the history.

**Comment by basz** [comment]
You get this? 

```
There know how to fix this warning?

1) ZfrOAuth2Test\Server\Grant\AuthorizationServerAwareTraitTest::testSetAuthorizationServer
Trying to configure method "setAuthorizationServer" which cannot be configured because it does not exist, has not been specified, is final, or is static
```

**Comment by prolic** [comment]
see: https://github.com/basz/zfr-oauth2-server/pull/1

**Comment by bakura10** [comment]
Side question: is there a way to deactivate all those Coveralls notification emails? Being quite spammed by those :(.

**Comment by basz** [comment]
Should be [here](https://coveralls.io/github/zf-fr/zfr-oauth2-server/settings), but i don't see what they [describe](https://coveralls.zendesk.com/hc/en-us/articles/201774895-Notifications)

I don't how to disable emails, but I did change the alert settings thresholds...  

```
If you would like to change the default notification settings for a repo, first go to the repository's detail view by clicking on the repo name from your repo index. On the detail view, you'll now see a nice, (not)shiny, new button labeled notifications:
```

**Comment by basz** [comment]
rebased to avoid merge conflict. i'm done with this, before 'minor thing' become 'not so minor' :-)

**Comment by prolic** [comment]
@basz please notify when ready for final review

**Comment by basz** [comment]
@prolic ready for final review (build was successful, coveralls was not (why?))

File: src/Exception/OAuth2Exception.php [hunk 6]
```diff
@@ -38,7 +41,7 @@ class OAuth2Exception extends Exception implements ExceptionInterface
      * @param string $message
```
Comment (prolic) [inline src/Exception/OAuth2Exception.php]: remove phpdoc

File: src/Model/AbstractToken.php [hunk 7]
```diff
@@ -71,7 +74,6 @@ private function __construct()
      * @param TokenOwnerInterface|null     $owner
      * @param Client|null                  $client
```
Comment (prolic) [inline src/Model/AbstractToken.php]: remove phpdoc

File: src/Model/Client.php [hunk 8]
```diff
@@ -95,7 +98,7 @@ public static function createNewClient(string $name, $redirectUris = null)
      * @param array $data
      * @return Client
```
Comment (prolic) [inline src/Model/Client.php]: remove phpdoc

File: src/Model/Scope.php [hunk 9]
```diff
@@ -80,9 +83,9 @@ public static function createNewScope(
 
     /**
      * @param array $data
-     * @return
+     * @return Scope
```
Comment (prolic) [inline src/Model/Scope.php]: remove phpdoc

File: src/ResourceServer.php [hunk 10]
```diff
@@ -59,7 +62,7 @@ public function __construct(AccessTokenService $accessTokenService)
      * @param  ServerRequestInterface $request
      * @param  array                  $scopes
      * @return AccessToken|null
-     * @throws Exception\InvalidAccessTokenException If given access token is invalid or expired
+     * @throws InvalidAccessTokenException If given access token is invalid or expired
      */
     public function getAccessToken(ServerRequestInterface $request, $scopes = [])
```
Comment (prolic) [inline src/ResourceServer.php]: array type hint

File: src/ResourceServer.php [hunk 11]
```diff
@@ -59,7 +62,7 @@ public function __construct(AccessTokenService $accessTokenService)
      * @param  ServerRequestInterface $request
      * @param  array                  $scopes
```
Comment (prolic) [inline src/ResourceServer.php]: remove phpdoc, only @throws needed here

File: src/ResourceServerInterface.php [hunk 12]
```diff
@@ -37,7 +41,7 @@
      * @param  ServerRequestInterface $request
      * @param  array|string|Scope[]   $scopes
      * @return AccessToken|null
-     * @throws Exception\InvalidAccessTokenException If given access token is invalid or expired
```
Comment (prolic) [inline src/ResourceServerInterface.php]: remove phpdoc, only @throws needed here

File: src/ResourceServerInterface.php [hunk 13]
```diff
@@ -37,7 +41,7 @@
...
+     * @throws InvalidAccessTokenException If given access token is invalid or expired
      */
     public function getAccessToken(ServerRequestInterface $request, $scopes = []);
```
Comment (prolic) [inline src/ResourceServerInterface.php]: array type hint

File: tests/src/Exception/OAuth2ExceptionTest.php [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+ * @covers  \ZfrOAuth2\Server\Exception\OAuth2Exception
+ */
+class OAuth2ExceptionTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @param $errorName
+     * @param $errorCode
+     * @dataProvider dataproviderErrorsCode
+     */
+    public function testErrorsCode($errorName, $expectedErrorCode)
```
Comment (prolic) [inline tests/src/Exception/OAuth2ExceptionTest.php]: type hints?

File: tests/src/Grant/AuthorizationGrantTest.php [hunk 15]
```diff
@@ -338,4 +338,17 @@ private function getValidAuthorizationCode($redirectUri = null, $owner = null, $
 
         return $token;
     }
+
+    public function testMethodGetType() {
```
Comment (prolic) [inline tests/src/Grant/AuthorizationGrantTest.php]: return void for all tests

File: tests/src/Grant/ClientCredentialsGrantTest.php [hunk 16]
```diff
@@ -73,15 +73,28 @@ public function testCanCreateTokenResponse()
...
+
+    public function testMethodGetType() {
+        $this->assertSame('client_credentials', $this->grant->getType());
+    }
+
+    public function testMethodGetResponseType() {
+        $this->assertSame('', $this->grant->getResponseType());
+    }
+
+    public function testMethodAllowPublicClients()
```
Comment (prolic) [inline tests/src/Grant/ClientCredentialsGrantTest.php]: return void

File: tests/src/Middleware/AuthorizationRequestMiddlewareTest.php [hunk 17]
```diff
@@ -52,20 +52,20 @@ public function setUp()
...
+        $this->authorizationServer->expects($this->once())
             ->method('handleAuthorizationRequest')
             ->with($request, $owner)
             ->willReturn($this->createMock(ResponseInterface::class));
 
         $middleware = $this->middleware;
         $middleware ($request, $response);
     }
-}
+}
```
Comment (prolic) [inline tests/src/Middleware/AuthorizationRequestMiddlewareTest.php]: missing new line at end of file

File: tests/src/Middleware/TokenRequestMiddlewareTest.php [hunk 18]
```diff
@@ -52,12 +52,12 @@ public function testCanHandleTokenRequest()
...
+        $this->authorizationServer->expects($this->once())
             ->method('handleTokenRequest')
             ->with($request)
             ->willReturn($this->createMock(ResponseInterface::class));
 
         $middleware = $this->middleware;
         $middleware ($request, $response, $next);
     }
-}
+}
```
Comment (prolic) [inline tests/src/Middleware/TokenRequestMiddlewareTest.php]: missing new line at end of file

File: tests/src/Model/RefreshTokenTest.php [hunk 19]
```diff
@@ -41,14 +40,24 @@ public function testGenerateNewAccessToken($ttl, $owner, $client, $scopes)
...
+        $this->assertSame($owner, $refreshToken->getOwner());
+
+        // with a ttl = 0, getExpiresAt must return null
+        if ($ttl === 0) {
+            $this->assertNull($refreshToken->getExpiresAt());
+        } else {
+            $this->assertInstanceOf(\DateTimeInterface::class, $refreshToken->getExpiresAt());
+            $this->assertEquals(
+                (new DateTimeImmutable())->modify("+$ttl seconds")->format(DateTime::ISO8601),
+                $refreshToken->getExpiresAt()->format(\DateTime::ISO8601)
```
Comment (prolic) [inline tests/src/Model/RefreshTokenTest.php]: use \DateTime::ATOM - in PHP ISO8601 is not ISO8601 format (lol)

File: tests/src/Model/RefreshTokenTest.php [hunk 20]
```diff
@@ -41,14 +40,24 @@ public function testGenerateNewAccessToken($ttl, $owner, $client, $scopes)
         /** @var RefreshToken $refreshToken */
         $refreshToken = RefreshToken::createNewRefreshToken($ttl, $owner, $client, $scopes);
 
-        $expiresAt = (new \DateTimeImmutable())->modify("+$ttl seconds");
+        $expiresAt = $ttl ? (new DateTimeImmutable())->modify("+$ttl seconds")->format(DateTime::ISO8601) : null;
```
Comment (prolic) [inline tests/src/Model/RefreshTokenTest.php]: use \DateTime::ATOM - in PHP ISO8601 is not ISO8601 format (lol)

File: src/AuthorizationServer.php [hunk 21]
```diff
@@ -72,10 +72,7 @@ class AuthorizationServer implements AuthorizationServerInterface
     private $refreshTokenService;
 
     /**
-     * @param ClientService       $clientService
-     * @param GrantInterface[]    $grants
-     * @param AccessTokenService  $accessTokenService
-     * @param RefreshTokenService $refreshTokenService
+     * @param GrantInterface[] $grants
```
Comment (prolic) [inline src/AuthorizationServer.php]: leaving only one param in the phpdoc makes it worse I think - all or nothing.

File: src/AuthorizationServer.php [hunk 22]
```diff
@@ -103,7 +100,6 @@ public function __construct(
     /**
      * Check if the authorization server supports this grant
      *
-     * @param  string $grantType
      * @return bool
```
Comment (prolic) [inline src/AuthorizationServer.php]: return bool is already in method signature

File: src/Grant/PasswordGrant.php [hunk 23]
```diff
@@ -144,17 +138,12 @@ public function createTokenResponse(
 
     /**
      * Set AuthorizationServer
```
Comment (prolic) [inline src/Grant/PasswordGrant.php]: isn't that description equals to the method name? so why not remove it :)

File: src/Model/AbstractToken.php [hunk 24]
```diff
@@ -70,9 +70,6 @@ private function __construct()
     /**
      * Create a new AbstractToken
      *
-     * @param int                          $ttl
-     * @param TokenOwnerInterface|null     $owner
-     * @param Client|null                  $client
      * @param string|string[]|Scope[]|null $scopes
```
Comment (prolic) [inline src/Model/AbstractToken.php]: same here, only one param doclblock left, makes it only worse, all or nothing.

File: src/Model/AuthorizationCode.php [hunk 25]
```diff
@@ -40,11 +40,7 @@ class AuthorizationCode extends AbstractToken
     /**
      * Create a new AuthorizationCode
      *
-     * @param int                          $ttl
-     * @param TokenOwnerInterface|null     $owner
-     * @param Client|null                  $client
      * @param string|string[]|Scope[]|null $scopes
```
Comment (prolic) [inline src/Model/AuthorizationCode.php]: same as above

File: src/Model/Client.php [hunk 26]
```diff
@@ -123,8 +115,6 @@ public function getId(): string
 
     /**
      * Get the client name
```
Comment (prolic) [inline src/Model/Client.php]: description = method name, so why description?

**Comment by prolic** [comment]
:fireworks: :+1:

**Pull request merged**